### PR TITLE
Integrate `lightning_utilities.core.rank_zero`

### PR DESCRIPTION
**Pull request by carmocca**
## What does this PR do?

Part of #14492 
~~Blocked by https://github.com/Lightning-AI/utilities/pull/36~~
Closes #14572

### Does your PR introduce any breaking changes? If yes, please list them.

None

## Before submitting

- [x] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/Lightning-AI/lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [x] Did you make sure to **update the documentation** with your changes? (if necessary)
- [x] Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- [x] Did you list all the **breaking changes** introduced by this pull request?
- [x] Did you **update the CHANGELOG**? (not for typos, docs, test updates, or minor internal changes/refactors)

## PR review

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

cc @borda @justusschock @awaelchli @rohitgr7

### CODE REVIEW

File: tests/tests_pytorch/utilities/test_rank_zero.py [hunk 1]
```diff
@@ -61,12 +62,3 @@ def test_rank_zero_priority(environ, expected_rank):
     """Test the priority in which the rank gets determined when multiple environment variables are available."""
     with mock.patch.dict(os.environ, environ):
         assert _get_rank() == expected_rank
-
-
-@pytest.mark.parametrize("env_vars", [{"RANK": "0"}, {"RANK": "1"}, {"RANK": "4"}])
-def test_rank_prefixed_message_with_env_vars(env_vars):
```
Comment (awaelchli) [inline tests/tests_pytorch/utilities/test_rank_zero.py:67]: I can't see where this moved. Is it in lightning-utilities?

**Review by awaelchli — APPROVED** [review]

File: tests/tests_pytorch/utilities/test_rank_zero.py (hunk 1 above)
Comment (carmocca) [reply tests/tests_pytorch/utilities/test_rank_zero.py:67]: Yes

Comment (carmocca) [reply tests/tests_pytorch/utilities/test_rank_zero.py:67]: https://github.com/Lightning-AI/utilities/blob/c27592abab2b24578753794c9bf005bc7f627595/tests/unittests/core/test_rank_zero.py#L12-L18

**Review by akihironitta — APPROVED** [review]

**Review by justusschock — APPROVED** [review]

**Review by Borda — APPROVED** [review]

**Review by tchaton — APPROVED** [review]
LGTM !

**Comment by awaelchli** [comment]
It is a miracle!

**Pull request merged**